### PR TITLE
Add support for workload name

### PR DESCRIPTION
**Pull request by hasit**
### Description of change
- If workload name is provided, use that as the index.
- Add validation for uniqueness of workload names

### Screenshot

screencapture-localhost-3000-d-0b57c46452c97079fe958a1b48bfd34a13e4ba25-aperture-service-protection-2023-05-26-10_04_38

##### Checklist

- [x] Tested in playground or other setup
- [x] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [x] Documentation is changed or added

### Summary by OpenAI

**New Feature:**
- Added support for workload names in Scheduler message and workload struct
- Implemented uniqueness validation for workload names

**Minor Change:**
- Removed an empty line in `pkg/metrics/schema.go`

> 🎉 A new feature we bring, with workload names to sing,
> Unique and validated, our scheduler's elevated,
> A minor change, an empty line no more,
> Together we celebrate, this PR we adore! 🚀

### CODE REVIEW

**Review by tanveergill — APPROVED** [review]

File: pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go [hunk 1]
```diff
@@ -520,13 +523,23 @@ func SanitizeSchedulerProto(proto *policylangv1.Scheduler) *policylangv1.Schedul
 		proto.DefaultWorkloadParameters = p
 	}
 
+	workloadNames := make(map[string]bool)
+
 	// Loop through the workloads
-	for _, workloadProto := range proto.Workloads {
+	for workloadIndex, workloadProto := range proto.Workloads {
+		if workloadProto.Name != "" {
+			if workloadNames[workloadProto.Name] {
```
Comment (tanveergill) [inline pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go:531]: Should add str(index) and default in workloadNames to check for all clashes

Comment (hasit) [reply pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go]: [556ff51](https://github.com/fluxninja/aperture/pull/2045/commits/556ff51fcd3f6ede4b336358952960cefdf394e3) addresses this.

File: pkg/policies/flowcontrol/actuators/load-scheduler/load-scheduler.go [hunk 2]
```diff
@@ -223,7 +223,10 @@ func (lsFactory *loadSchedulerFactory) newLoadSchedulerOptions(
 		logger.Warn().Err(err).Msg("Failed to unmarshal load scheduler config wrapper")
 		return fx.Options(), err
 	}
-	loadSchedulerProto.Parameters.Scheduler = workloadscheduler.SanitizeSchedulerProto(loadSchedulerProto.Parameters.Scheduler)
+	loadSchedulerProto.Parameters.Scheduler, err = workloadscheduler.SanitizeSchedulerProto(loadSchedulerProto.Parameters.Scheduler)
+	if err != nil {
+		return fx.Options(), err
+	}
```
[the remark this answers is not in the document]
Comment (hasit) [reply pkg/policies/flowcontrol/actuators/load-scheduler/load-scheduler.go]: [b80ee1d](https://github.com/fluxninja/aperture/pull/2045/commits/b80ee1d7a9fc3d016a1ce6b2734db95b06e8c93b) addresses this.

File: pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go [hunk 3]
```diff
@@ -353,17 +351,22 @@ func (ws *Scheduler) Decide(ctx context.Context,
...
+		matchedWorkloadParametersProto = ws.proto.DefaultWorkloadParameters
 		matchedWorkloadIndex = metrics.DefaultWorkloadIndex
 	}
 
 	fairnessLabel := "workload:" + matchedWorkloadIndex
 
-	if val, ok := labels[matchedWorkloadProto.FairnessKey]; ok {
+	if val, ok := labels[matchedWorkloadParametersProto.FairnessKey]; ok {
 		fairnessLabel = fairnessLabel + "," + val
 	}
```
[the remark this answers is not in the document]
Comment (hasit) [reply pkg/policies/flowcontrol/actuators/workload-scheduler/scheduler.go:371]: I have already added a step `SanitizeSchedulerProto` function to ensure that all workload names are unique.

**Pull request merged**